src/spyglass/decoding/v0/utils.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+import xarray as xr
+
+from spyglass.utils import logger
+
+try:
+    from replay_trajectory_classification.classifier import (
+        _DEFAULT_CLUSTERLESS_MODEL_KWARGS,
+        _DEFAULT_CONTINUOUS_TRANSITIONS,
+        _DEFAULT_ENVIRONMENT,
+        _DEFAULT_SORTED_SPIKES_MODEL_KWARGS,
+    )
+    from replay_trajectory_classification.discrete_state_transitions import (
+        DiagonalDiscrete,
+    )
+    from replay_trajectory_classification.initial_conditions import (
+        UniformInitialConditions,
+    )
+except (ImportError, ModuleNotFoundError) as e:
+    (
+        _DEFAULT_CLUSTERLESS_MODEL_KWARGS,
+        _DEFAULT_CONTINUOUS_TRANSITIONS,
+        _DEFAULT_ENVIRONMENT,
+        _DEFAULT_SORTED_SPIKES_MODEL_KWARGS,
+        DiagonalDiscrete,
+        UniformInitialConditions,
+    ) = [None] * 6
+    logger.warning(e)
+
+
+def get_time_bins_from_interval(interval_times: np.array, sampling_rate: int):
+    """Gets the superset of the interval."""
+    start_time, end_time = interval_times[0][0], interval_times[-1][-1]
+    n_samples = int(np.ceil((end_time - start_time) * sampling_rate)) + 1
+
+    return np.linspace(start_time, end_time, n_samples)
+
+
+def discretize_and_trim(series: xr.DataArray, ndims=2) -> xr.DataArray:
+    """Discretizes a continuous series and trims the zeros.
+
+    Parameters
+    ----------
+    series : xr.DataArray, shape (n_time, n_position_bins)
+        Continuous series to be discretized
+    ndims : int, optional
+        Number of dimensions of the series. Default is 2 for 1D series (time,
+        position), 3 for 2D series (time, y_position, x_position)
+
+    Returns
+    -------
+    discretized : xr.DataArray, shape (n_time, n_position_bins)
+        Discretized and trimmed series
+    """
+    index = (
+        ["time", "position"]
+        if ndims == 2
```

Comment:
i think this should be `if ndims ==1` (?)

Reply:
On seeing the calls to this function, this would run correctly.  However, my first assumption as a user was that `ndims` referred to the number of spatial dimensions (e.g. 1 or 2).  

If change to that preference need to make the calls change as well